src/sentry/static/sentry/app/views/settings/projectDataPrivacy/projectDataPrivacyRulesPanel.tsx

```diff
@@ -0,0 +1,214 @@
+import styled from '@emotion/styled';
+
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import {Panel, PanelHeader, PanelBody} from 'app/components/panels';
+import Link from 'app/components/links/link';
+import {IconAdd} from 'app/icons/iconAdd';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {addMessage} from 'app/actionCreators/indicator';
+
+import ProjectDataPrivacyRulesForm from './projectDataPrivacyRulesForm';
+import {DATA_TYPE, ACTION_TYPE} from './utils';
+
+type Rule = Omit<
+  React.ComponentProps<typeof ProjectDataPrivacyRulesForm>,
+  'onDelete' | 'onChange'
+>;
+
+type State = {
+  rules: Array<Rule>;
+  savedRules: Array<Rule>;
+  status?: 'success' | 'loading' | 'error';
+};
+
+let requestResponse: Array<Rule> = [
+  {
+    id: 1,
+    action: ACTION_TYPE.MASK,
+    data: DATA_TYPE.BANK_ACCOUNTS,
+    from: 'api_key && !$object',
+  },
+  {
+    id: 2,
+    action: ACTION_TYPE.REMOVE,
+    data: DATA_TYPE.IP_ADDRESSES,
+    from: 'xxx && xxx',
+  },
+];
+
+class ProjectDataPrivacyRulesPanel extends React.Component<{}, State> {
+  state: State = {
+    rules: [],
+    savedRules: [],
+    status: 'loading',
+  };
+
+  componentDidMount() {
+    this.loadRules();
+  }
+
+  loadRules = async () => {
+    // add request here
+    const result: Array<Rule> = await new Promise(resolve => {
+      setTimeout(async function() {
+        resolve(requestResponse);
+      }, 1000);
+    });
+
+    this.setState({
+      rules: result,
+      savedRules: result,
+      status: undefined,
+    });
+  };
+
+  formRef: React.RefObject<HTMLFormElement> = React.createRef();
+
+  handleAddRule = () => {
+    this.setState(prevState => ({
+      rules: [
+        ...prevState.rules,
+        {
+          id: prevState.rules.length + 1,
+          action: ACTION_TYPE.MASK,
+          data: DATA_TYPE.BANK_ACCOUNTS,
+          from: '',
+        },
+      ],
+    }));
+  };
+
+  handleDeleteRule = (ruleId: number) => () => {
+    this.setState(prevState => ({
+      rules: prevState.rules.filter(rule => rule.id !== ruleId),
+    }));
+  };
+
+  handleChange = (updatedRule: Rule) => {
+    this.setState(prevState => ({
+      ...prevState,
+      rules: prevState.rules.map(rule => {
+        if (rule.id === updatedRule.id) {
+          return updatedRule;
+        }
+        return rule;
+      }),
+    }));
+  };
+
+  handleSubmit = () => {
+    this.setState(
+      prevState => ({
+        status: 'success',
+        savedRules: prevState.rules,
+      }),
+      () => {
+        requestResponse = this.state.rules;
+      }
+    );
+  };
+
+  handleValidation = () => {
+    const {rules} = this.state;
+    const isAnyRuleFieldEmpty = rules.find(rule =>
+      Object.keys(rule).find(ruleKey => !rule[ruleKey])
+    );
+
+    const isFormValid = !isAnyRuleFieldEmpty;
+
+    if (isFormValid) {
+      this.formRef.current?.dispatchEvent(new Event('submit'));
```

Comment:
Can't we just call handleSubmit here?

Reply:
yes!! :D the initial idea was to get the data by sending the form, but things have changed and I forgot to change this logic ...updating it in 1...2....3...